### PR TITLE
Bump `fuel-vm` to `0.35.0` version

### PR DESCRIPTION
**Pull request by xgreenx**
The PR uses the latest `fuel-vm 0.35.0`. It affects the `fuel-core` in the next way:
- Removes support for `OpaqueReceipt` transaction
- Adds new opcodes `eck1`, `ecr1`, `ed19`
- Adds new `Receipt::Mint` and `Receipt::Burn`

### CODE REVIEW

File: benches/benches/set/crypto.rs [hunk 1]
```diff
@@ -20,9 +20,9 @@ pub fn run(c: &mut Criterion) {
     let signature = Signature::sign(&secret, &message);
 
     run_group_ref(
```
Comment (Voxelot) [inline benches/benches/set/crypto.rs:22]: should we also setup benchmarks for ecr1 and ed19?

Comment (xgreenx) [reply benches/benches/set/crypto.rs:22]: There is an issue for that=) https://github.com/FuelLabs/fuel-core/issues/1254

File: crates/fuel-core/src/query/message.rs [hunk 2]
```diff
@@ -164,6 +164,7 @@ pub fn message_proof<T: MessageProofData + ?Sized>(
         Some(r) => r,
         None => return Ok(None),
     };
+    let data = data.ok_or(anyhow::anyhow!("Receipt doesn't have a `data`"))?;
```
Comment (Voxelot) [inline crates/fuel-core/src/query/message.rs:167]:
```suggestion
    let data = data.ok_or(anyhow::anyhow!("Output message doesn't contain any `data`"))?;
```

File: tests/tests/tx_gossip.rs [hunk 3]
```diff
@@ -76,8 +76,7 @@ async fn test_tx_gossiping() {
         .next()
         .expect("Producer is initialized with one transaction");
 
-    let response = client_one.transaction(&tx_id).await.unwrap();
-    assert!(response.is_some());
+    let _ = client_one.await_transaction_commit(&tx_id).await.unwrap();
```
Comment (Voxelot) [inline tests/tests/tx_gossip.rs:79]: is this a workaround for the flakey test?

**Review by Voxelot — APPROVED** [review]

**Review by leviathanbeak — APPROVED** [review]

**Pull request merged**